enterprise/micronaut/src/org/netbeans/modules/micronaut/completion/MicronautDataCompletionCollector.java

```diff
@@ -53,6 +53,7 @@ public Completion createSQLItem(CompletionItem item) {
                 return CompletionCollector.newBuilder(item.getInsertPrefix().toString())
                         .insertText(item.getInsertPrefix().toString().replace("\"", "\\\""))
                         .kind(Completion.Kind.Property)
+                        .sortText(String.format("%04d%s", item.getSortPriority(), item.getSortText()))
```

Comment:
This is really ugly. The moment someone decides to report a priority of 10000 this will become a mess to debug. Given that this can't be easily changes as it seems to be baked into the protocol, I suggest to encode it like this:

```suggestion
                        .sortText(String.format("%010d%s",
                                Long.valueOf(item.getSortPriority()) + Math.abs(Long.valueOf(Integer.MIN_VALUE)),
                                item.getSortText())
                        )
```

The shift of  `Math.abs(Long.valueOf(Integer.MIN_VALUE))` ensures we get only positive values and using 10 digits allows to encode all possible values